### PR TITLE
Initial SDK Review 

### PR DESCRIPTION
**Pull request by barrettk8090**
This is an initial, large-ish PR for the DIMO Python SDK. Currently, it does not yet include support for Streamr, nor tests/detailed error messages, but those are planned for a later date. It is also not yet fully configured for package upload to PyPi (currently working thru testing this on TestPyPi), but those changes will be made once the initial SDK has been reviewed and approved by the DIMO team. 

**How To Test (for now, pre-PyPi):**
1. Clone the repo locally and switch to the `dev-barrettk` branch & pull.
2. In your terminal, `cd` into the `dimo` folder
3. Create a new file called `test.py` inside the `dimo` folder and import the following (as needed) and initialize dimo
```python
from dimo import DIMO
import asyncio
from dotenv import load_dotenv # Only needed if planning to add a private key into your .env for testing
import os # Only needed if planning to add a private key into your .env for testing 

dimo = DIMO("Production")
```
4. Inside `test.py`, create an async method following this format (see considerations note below):    
```python
async def main():

  #Your test code here

if __name__ == "__main__":
        asyncio.run(main())
```
5. Use the README to follow along/test different endpoints. You can run your file in the terminal by running `python test.py`. If you're using a `.env` for your private key, and you're testing the `get_token()` method, your code might look something like this:

```python
async def main():

    authHeader = await dimo.auth.get_token(
                client_id='<client_id>',
                domain='<domain>',
                private_key=os.getenv("test_private_key")
    )
    print("Full Auth Header: ", authHeader)
    access_token = authHeader["access_token"]
    print("My access_token: ", access_token)

if __name__ == "__main__":
        asyncio.run(main())
```

**Considerations/Notes:**
* Currently, the SDK uses [asyncio](https://docs.python.org/3/library/asyncio.html) to make asynchronous requests for all endpoints, in an attempt to generally follow the same patterns as the NodeJS SDK. This especially makes sense for methods such as `get_token()`, but may not be required for other endpoints. Would love to get thoughts/feedback on this, as I’d like to make the SDK as accessible/easy to use for devs as possible. 
* The SDK currently uses the same template graphQL queries for the Identity & Telemetry classes as the NodeJS SDK, but more can be added. 
* README is mostly complete with some placeholders on items that are “Coming Soon”. Note that the pip installation is also placeholder and is not yet functional.

### CODE REVIEW

File: .gitignore
Comment (0xsemaj) [inline .gitignore:1]: Probably better to include the distribution files, logs, and other cache files.

https://stackoverflow.com/questions/3719243/best-practices-for-adding-gitignore-file-for-python-projects

File: README.md [hunk 1]
```diff
@@ -1 +1,368 @@
...
+
+or
+
+```python
+dimo = DIMO("Dev")
+```
+
+### Developer License
+
+As part of the authentication process, you will need to register a set of `client_id` and `redirect_uri` (aka `domain`) on the DIMO Network. The [DIMO Developer License](https://docs.dimo.zone/developer-platform/getting-started/developer-license) is our approach and design to a more secured, decentralized access control. As a developer, you will need to perform the following steps:
```
Comment (0xsemaj) [inline README.md:41]: Should update to console.dimo.xyz for Dev License registration. It'll simplify the whole process

File: README.md [hunk 2]
```diff
@@ -1 +1,368 @@
...
+### Querying the DIMO GraphQL API
+
+The SDK accepts any type of valid custom GraphQL queries, but we've also included a few sample queries to help you understand the DIMO GraphQL APIs.
+
+#### Authentication for GraphQL API
+
+The GraphQL entry points are designed almost identical to the REST API entry points. For any GraphQL API that requires auth headers (Telemetry API for example), you can use the same pattern as you would in the REST protected endpoints.
+
+```python
+privilege_token = await dimo.token_exchange.exchange(access_token, privileges=[1,3,4], token_id=<vehicle_token_id>)
```
Comment (0xsemaj) [inline README.md:168]: This endpoint seems odd given that `tokens` are passed thru the `token` argument in the other endpoints. Might want to take a more uniform approach, either `token` or `access_token/privilege_token`

File: dimo/config.py
Comment (0xsemaj) [inline dimo/config.py:1]: What is this file for?

File: pyproject.toml
Comment (0xsemaj) [inline pyproject.toml:1]: We prefer using Apache 2.0 license FYI: https://github.com/DIMO-Network/dimo-node-sdk/blob/master/LICENSE

File: dimo/auth.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    # Requires client_id, domain, and private_key. Address defaults to client_id.
+    async def get_token(self, 
+                        client_id, 
+                        domain, 
+                        private_key, 
+                        address=None, 
+                        scope='openid email', 
+                        response_type='code', 
+                        grant_type="authorization_code", 
+                        env="Production"):
```
Comment (0xsemaj) [inline dimo/auth.py:52]: Why is this hard coded to Production?

**Review by jamesliupenn — APPROVED** [review]
Minor questions on some of the setup but overall this is a good start to be published on PyPI. We may need to rework some of the structure as configuration doesn't seem to be easy. Will also need to include a contribution guide and add in Attestation API

**Pull request merged**